packages/core/src/State.ts

```diff
@@ -132,13 +141,17 @@ export class State<
    * @param stateValue
    * @param context
    */
-  public static from<TC, TE extends EventObject = EventObject>(
-    stateValue: State<TC, TE> | StateValue,
+  public static from<
+    TC,
+    TE extends EventObject = EventObject,
+    TA extends { type: string } = { type: string; [key: string]: any }
```

Comment:
Side-thought: might need to include `TTypestate` before `TA`, ran into the need for it, as it's missing from here